farm.js

```diff
@@ -8,24 +8,30 @@ const feedCow = require('cow-foo/feed-cow');
 const feedBird = require('./lib/feed-bird');
 const eatLunch = require('./lib/eat-lunch');
 
+/** @todo: Fix feedCow to use Promises and throw away this wrapper */
+const feedCowAsPromise = cow=> {
+  return new Promise(resolve=> {
+    feedCow(cow, resolve);
+  });
+};
 
 function main(cb) {
   const cow = new Cow();
-  feedCow(cow, function() {
-    var bird = new Bird();
-    var birdFood = (bird == 'pigeon') ? 'seed' : 'corn';
-    feedBird(bird, birdFood, function() {
-      eatLunch(function() {
-        cb();
-      });
-    });
-  });
+
+  return feedCowAsPromise(cow).
+    then(()=> {
+      const bird = new Bird();
+      const birdFood = (bird == 'pigeon') ? 'seed' : 'corn';
+
+      feedBird(bird, birdFood);
+    }).
+    then(()=> eatLunch());
 }
 
 if (require.main === module) {
-  main(function(){
+  main().then(()=> {
```

Comment:
Because main is exported, at after refactoring it doesn't accept callback and returns another value (Promise instead of undefined). First could be a problem...